src/os/src/detail/osTime.c

```diff
@@ -557,15 +557,23 @@ int64_t taosTimeSub(int64_t timeStamp, int64_t duration, char unit, int32_t prec
   return (int64_t)(mktime(&tm) * TSDB_TICK_PER_SECOND(precision));
 }
 
-
 int32_t taosTimeCountInterval(int64_t skey, int64_t ekey, int64_t interval, char unit, int32_t precision) {
   if (ekey < skey) {
     int64_t tmp = ekey;
     ekey = skey;
     skey = tmp;
   }
+
+#ifdef _MSC_VER
+#if _MSC_VER >= 1900
+  int64_t timezone = _timezone;
+#endif
+#endif
+
+  int64_t tz_offset = -1 * timezone * TSDB_TICK_PER_SECOND(precision);
+
   if (unit != 'n' && unit != 'y') {
-    return (int32_t)((ekey - skey) / interval);
```

Comment:
The old computation is just to compute the length of the time range of [s, e] divided by the size of interval, rather than the number of covering aligned time intervals as required. So it caused violation of an assertion.